Makefile

```diff
@@ -1,31 +1,27 @@
-help: ## Display this help screen
-	@grep -h -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-15s\033[0m %s\n", $$1, $$2}'
+.PHONY: wasm contracts keys circuits abi test run help
 
-wasm: ## Generate the WASM for the contract given (e.g. make wasm for=transfer)
-	@cargo rustc \
-		--manifest-path=contracts/$(for)/Cargo.toml \
-		--release \
-		--target wasm32-unknown-unknown \
-		-- -C link-args=-s
+wasm: ## Generate the WASM for all the contracts
+	$(MAKE) -C ./contracts/transfer/ wasm
+	$(MAKE) -C ./contracts/bid/ wasm
 
-contracts: ## Generate the WASM for all the contracts & test them
-	$(MAKE) -C ./contracts/transfer/ test
-	$(MAKE) -C ./contracts/bid/ test
+abi: ## Build the ABI and test it
+	$(MAKE) -C ./rusk-abi test
 
 keys: ## Create the keys for the circuits
 	cd rusk && cargo test --release -vv -- --nocapture
 
+contracts: ## Execute the test for all contracts
+	$(MAKE) -C ./contracts/transfer/ test
+	$(MAKE) -C ./contracts/bid/ test
+
 circuits: ## Build and test circuit crates
```

Comment:
This also should be updated as well. In my opinion the root `Makefile` should call the `Makefile` of the subfolder (if any) and execute with the same recipe. In that case `contracts` folder should have a Makefile that deals with the contracts, where the root Makefile should only invoke the Makefile of the contracts folder (not sure if I'm clear)